ref/context_sha2.c

```diff
@@ -0,0 +1,43 @@
+#include "context.h"
+
+/**
+ * Absorb the constant pub_seed using one round of the compression function
+ * This initializes state_seeded and state_seeded_512, which can then be
+ * reused in thash
+ **/
+static void seed_state(spx_ctx *ctx) {
+    uint8_t block[SPX_SHA512_BLOCK_BYTES];
+    size_t i;
+
+    for (i = 0; i < SPX_N; ++i) {
+        block[i] = ctx->pub_seed[i];
+    }
+    for (i = SPX_N; i < SPX_SHA512_BLOCK_BYTES; ++i) {
+        block[i] = 0;
+    }
+    /* block has been properly initialized for both SHA-256 and SHA-512 */
+
+    sha256_inc_init(&ctx->state_seeded);
+    sha256_inc_blocks(&ctx->state_seeded, block, 1);
+#if SPX_SHA512
+    sha512_inc_init(&ctx->state_seeded_512);
+    sha512_inc_blocks(&ctx->state_seeded_512, block, 1);
+#endif
+}
+
+
+/* For SHA, there is no immediate reason to initialize at the start,
```

Comment:
This is confusing to me, the comment says it's an empty operation, but it actually calls `seed_state` which looks like it does stuff.

Reply:
🤔  I copied this without thinking too much about the comments from `hash_sha2.c`.